query-algebrizer/src/clauses/mod.rs

```diff
@@ -108,6 +109,22 @@ impl<K: Clone + Ord, V: Clone> Intersection<K> for BTreeMap<K, V> {
             .filter_map(|(k, v)| ks.when_contains(k, || (k.clone(), v.clone())))
             .collect()
     }
+
+    /// Remove all keys from the map that are not present in `ks`.
+    /// This implementation is terrible because there's no mutable iterator for BTreeMap.
+    fn keep_intersected_keys(&mut self, ks: &BTreeSet<K>) {
+        let mut to_remove = Vec::with_capacity(self.len() - ks.len());
+        {
```

Comment:
Annotate scope braces with what they're holding/dropping, and is there a reason to not indent?